File: backend/php/laravel/app/Http/Controllers/PostController.php
<?php

namespace  App\Http\Controllers;

use dao\PostDAO;
use Illuminate\Http\Request;
use exception\IllegalArgumentException;
use Illuminate\Http\Response;
use model\Post;

class PostController extends Controller {

    /**
     * @var PostDAO
     */
    private $dao;

    public function __construct(PostDAO $dao)
    {
        $this->dao = $dao;
    }

    function fetchAll() {
        return $this->dao->findAll();
    }

    function createPost(Request $request) {
        try {
            $content = $request->getContent();
            $decodedContent = json_decode($content);
            $post = Post::fromArray($decodedContent);
            $this->dao->create($post);
            return new Response(null, Response::HTTP_NO_CONTENT);
        } catch (IllegalArgumentException $e) {
            return new Response(null, Response::HTTP_BAD_REQUEST);
        }
    }

    function updatePost($id, Request $request) {
        try {
            $content = $request->getContent();
            $decodedContent = json_decode($content);
            $post = Post::fromArray($decodedContent);

            if($post->getId() != $id) {
                return new Response(null, Response::HTTP_BAD_REQUEST);
            }

            $this->dao->update($post);
            return new Response(null, Response::HTTP_NO_CONTENT);
        } catch (IllegalArgumentException $e) {
            return new Response(null, Response::HTTP_NOT_FOUND);
        }
    }


    function deletePost($id) {
        try {
            $this->dao->delete($id);
            return new Response(null, Response::HTTP_NO_CONTENT);
        } catch (IllegalArgumentException $e) {
            return new Response(null, Response::HTTP_NOT_FOUND);
        }
    }
}

File: backend/csharp/asp-net/Controllers/PostController.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Threading.Tasks;
using Microsoft.AspNetCore.Mvc;
using dao;
using dao.models;
using System.Net.Http;

namespace asp_net.Controllers {
  [ApiController]
  [Route("blog")]
  public class PostController : ControllerBase {
    private PostDAO dao;

    public PostController(PostDAO dao) {
      this.dao = dao;
    }

    [HttpGet]
    [Route("post")]
    public IEnumerable<Post> FetchAllPosts() {
      var findAllTask = this.dao.FindAll();
      findAllTask.Start();
      findAllTask.Wait();
      return findAllTask.Result;
    }

    [HttpPost]
    [Route("post")]
    public ActionResult CreatePost([FromBody]Post post) {
      try {
        this.dao.Create(post);
        return NoContent();
      } catch (Exception e) {
        return BadRequest();
      }
    }

    [HttpPut]
    [Route("{postId}")]
    public ActionResult UpdatePost([FromRoute]int postId, [FromBody]Post post) {
      if (postId != post.Id) {
        return BadRequest();
      }

      try {
        this.dao.Update(post);
        return NoContent();
      } catch (Exception e) {
        return NotFound();
      }
    }

    [HttpPost]
    [Route("post")]
    public ActionResult DeletePost([FromRoute]int postId) {
      try {
        this.dao.Delete(postId);
        return NoContent();
      } catch (Exception e) {
        return NotFound();
      }
    }

  }
}
